PWGHF/D2H/Tasks/taskDplus.cxx

```diff
@@ -275,17 +451,34 @@ struct HfTaskDplus {
         if ((yCandRecoMax >= 0. && std::abs(hfHelper.yDplus(candidate)) > yCandRecoMax)) {
           continue;
         }
+
+        if (storeCentrality || storeOccupancy) {
+          auto collision = candidate.template collision_as<CollisionsCent>();
+          if (storeCentrality && centEstimator != CentralityEstimator::None) {
+            cent = getCentrality(collision);
+          }
+          if (storeOccupancy && occEstimator != OccupancyEstimator::None) {
+            occ = getOccupancy(collision);
+          }
+        }
+
         fillHisto(candidate);
-        fillSparseML<false, false>(candidate);
+        fillSparseML<false, false>(candidate, ptBhad, flagBHad, cent, occ);
       }
     }
   }
+
   // Run analysis for the reconstructed Dplus candidates with MC matching
-  /// \param candidates are reconstructed candidates
-  /// \param mcParticles are particles with MC information
-  template <bool fillMl, typename T1, typename T2>
-  void runMCAnalysis(const T1& /*recoCandidates*/, const T2& mcParticles)
+  /// \param recoCandidates are reconstructed candidates
+  /// \param recoColls are reconstructed collisions
+  template <bool fillMl, typename T1>
+  void runMCRecoAnalysis(const T1& /*recoCandidates*/, McRecoCollisionsCent const& /*recoColls*/)
```

Comment:
Where is `T1` needed?

Reply:
This function is independent of the process functions and neither `T1` nor `recoCandidates` are used anywhere in it.